common/src/main/java/net/opentsdb/query/readcache/CachedQueryNode.java

```diff
@@ -136,6 +137,22 @@ public Builder toBuilder() {
     return null;
   }
 
+  @Override
+  public List<QueryResultId> resultIds() {
+    // TODO - do we need to populate this?
+    return Collections.emptyList();
```

Comment:
why not return a null. I understand, the client will have to do a null check. But, it will avoid the object creation.

Reply:
It uses the static empty list here.

Reply:
You are right. That should be fine then.